File: Practice/src/introduction_coding_tests/FindModeValue.py
def solution(array):
    answer = 0
    arr = dict()
    max = 0
    for i in array:
        count = arr.get(i, 0) + 1
        arr[i] = count
        if count > max:
            max = count
            answer = i
        elif count == max:
            answer = -1
        print(arr)
    return answer

File: Practice/src/introduction_coding_tests/FindModeValue.js
function solution(array) {
    var answer = 0;
    const m = new Map();
    let max = 0;
    for (let i of array) {
        let count = m.has(i) ? m.get(i) + 1 : 1;
        m.set(i, count);
        if (max < count) {
            answer = i;
            max = count;
        } else if (max == count) {
            answer = -1;
        }
    }
    return answer;
}
